File: 455.分发饼干.java
import java.util.*;
/*
 * @lc app=leetcode.cn id=455 lang=java
 *
 * [455] 分发饼干
 */

// @lc code=start
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int ans = 0, cookie = 0;
        while (ans < g.length && cookie < s.length) {
            if (g[ans] <= s[cookie]) {
                ++ans;
            } // else current cookie does not satisfy
            ++cookie;
        }
        return ans;
    }
}
// @lc code=end

File: 455.分发饼干.py
#
# @lc app=leetcode.cn id=455 lang=python3
#
# [455] 分发饼干
#

# @lc code=start
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort()
        s.sort()
        child = cookie = 0
        while child < len(g) and cookie < len(s):
            # if g[child] <= s[cookie]:
            #     child += 1
            child += (g[child] <= s[cookie])
            cookie += 1
        return child

# @lc code=end

